api/Dockerfile

```diff
@@ -1,14 +1,39 @@
-FROM node:18.15.0-alpine
+##################
+### BASE IMAGE ###
+##################
+FROM node:18.15.0-alpine AS base
+
+# Directory used in container
+WORKDIR /usr/api/
+
+# Copy dependency package, install, copy everything else
+COPY package*.json .
+RUN npm i --omit=dev
+COPY . .
+
+# Install TypeScript. Needed for build process.
+RUN npm i -D typescript
```

Comment:
I recommend using a specific version of typescript like typescript@4.9.5 or typescript@5.0.4

Reply:
Local version is 5.0.4, so went with that.